core/src/test/java/com/netflix/iceberg/TestBaseTableScan.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg;
+
+import com.netflix.iceberg.types.Types;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+
+import static com.netflix.iceberg.types.Types.NestedField.required;
+import static org.junit.Assert.assertEquals;
+
+public class TestBaseTableScan {
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  private final Schema schema = new Schema(
+    required(1, "id", Types.IntegerType.get()),
+    required(2, "data", Types.StringType.get())
+  );
+  private File tableDir = null;
+
+  @Before
+  public void setupTableDir() throws IOException {
+    this.tableDir = temp.newFolder();
+  }
+
+  @After
+  public void cleanupTables() {
+    TestTables.clearTables();
+  }
+
+  @Test
+  public void testTableScanHonorsSelect() {
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Table table = TestTables.create(tableDir, "test", schema, spec);
+
+    TableScan scan = table.newScan().select("id");
+
+    Schema expectedSchema = new Schema(
+      required(1, "id", Types.IntegerType.get())
+    );
+
+    assertEquals("A tableScan.select() should prune the schema",
+      expectedSchema.asStruct(),
+      scan.schema().asStruct()
+    );
+  }
+
+  @Ignore
+  @Test
+  public void testTableScanHonorsSelectWithoutCaseSensitivity() {
```

Comment:
While cleaning up this PR I noticed I missed making case sensitivity apply to selected columns. This test case shows the issue.

Will fix this shortly in this PR.

Reply:
Fixed.